distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP.enso

```diff
@@ -407,66 +146,112 @@ type HTTP
 
              import Standard.Base.Data.Time.Duration.Duration
              import Standard.Base.Network.HTTP.HTTP
-             import Standard.Base.Network.HTTP.Form.Form
              import Standard.Base.Network.HTTP.HTTP_Method.HTTP_Method
              import Standard.Base.Network.HTTP.Request.Request
 
              example_request =
-                 form = [Form.text_field "name" "John Doe"]
+                 form = Map.from_vector [["name", "John Doe"]]
                  req = Request.new HTTP_Method.Post "http://httpbin.org/post" . with_form form
                  http = HTTP.new (timeout = (Duration.new seconds=30))
                  http.request req
-    request : Request -> Response ! Request_Error
-    request self req =
+
+       > Example
+         Send a Get request with headers, and auto-parse the result
+         Note: This example will send a network request.
+
+             import Standard.Base.Network.HTTP.HTTP
+             import Standard.Base.Network.HTTP.HTTP_Method.HTTP_Method
+             import Standard.Base.Network.HTTP.Request.Request
+
+             example_request =
+                 req = Request.new HTTP_Method.Get "http://httpbin.org/get" try_auto_parse_response=True
+                 res = Examples.http_client.request req
+                 res.body
+    request : Request -> Boolean -> Response ! Request_Error
+    request self req try_auto_parse_response=False =
+        check_output_context ~action =
+            if Context.Output.is_enabled then action else
+                Error.throw (Forbidden_Operation.Error "HTTP POST/PUT/PATCH are forbidden as the Output context is disabled.")
```

Comment:
What about GET? Is it allowed?

And DELETE is probably disallowed too and not on this list.

Reply:
Also it seems that this check runs for GET as well. Or am I misreading something?

Reply:
Done.
It does; since GET endpoints are free to do whatever they want, it makes sense to me to prevent all of them when the output context is disabled. @jdunkerley?

Reply:
From a rationalistic perspective, I agree with you and that's my personal preference (no requests at all with Output disabled) - seems safer.

But the pragmatic perspective suggests that GET should be allowed even with Output disabled. That is because in practice it will be used as inputs a lot and making it not run will be a huge pain. And while GET can do whatever, most sane GETs should be idempotent and rather safe to run.

We have the `Context.Input` that I think fits the GET use case perfectly - we just haven't integrated it with the IDE fully yet (and I think it may need some caching to work better). So for now I'd keep GET unconstrained and in the future we will probably add it to Input context.